vllm/engine/llm_engine.py

```diff
@@ -144,9 +145,8 @@ def __init__(
         self.log_stats = log_stats
 
         if not self.model_config.skip_tokenizer_init:
-            self.tokenizer: BaseTokenizerGroup
-            self._init_tokenizer()
-            self.detokenizer = Detokenizer(self.tokenizer)
+            tokenizer = self._init_tokenizer()
+            self.detokenizer = Detokenizer(tokenizer)
```

Comment:
I'm developing on VSCode, and have noticed that Pyright also applies the annotation to the `else` branch, resulting in a type error since `self.tokenizer` is set to `None` in that case. This change makes it so that `self.tokenizer` implicitly has a type of `Optional[BaseTokenizerGroup]`.